### PR TITLE
STORAGE/Refactor & Testing

### PR DESCRIPTION
**Pull request by boonwj**
This PR brings the following changes to the HoloStorageAccessor

## Refactor
- Shift configuration and blob initialisation out of the router setup function.
- Configuration will be passed in to the setup function

## Tests
- Create mock FHIR server for testing of responses 
- Add tests for all GET and PUT endpoints in HoloStorageAccessor 

Code coverage currently at ~70%. Lacking tests for blobstorage related code (e.g. Hologram DELETE / POST, blobstorage operations)

### CODE REVIEW

File: HoloStorageAccessor/internal/holostorageaccessor/api_default_test.go [hunk 1]
```diff
@@ -0,0 +1,1470 @@
...
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+// This creates a test server that mocks the FHIR server responses
```
Comment (nielsboecker) [inline HoloStorageAccessor/internal/holostorageaccessor/api_default_test.go]: Maybe a separate file?

File: HoloStorageAccessor/internal/holostorageaccessor/api_default_test.go [hunk 2]
```diff
@@ -0,0 +1,1470 @@
...
+							"given": [
+								"Britney"
+							],
+							"prefix": [
+								"Ms"
+							]
+						}
+					],
+					"gender": "female",
+					"birthDate": "1960-01-02"
```
Comment (nielsboecker) [inline HoloStorageAccessor/internal/holostorageaccessor/api_default_test.go]: Damn Britney is old

File: HoloStorageAccessor/internal/holostorageaccessor/api_default_test.go [hunk 3]
```diff
@@ -0,0 +1,1470 @@
...
+	}
+
+	tests := map[string]test{
+		"no_aids_query": {
+			inUrl:      "/api/v1/patients",
+			inMethod:   http.MethodGet,
+			wantStatus: 400,
+		},
+		"wrong_query_field": {
+			inUrl:      "/api/v1/patients?pid=p1000",
```
Comment (nielsboecker) [inline HoloStorageAccessor/internal/holostorageaccessor/api_default_test.go:221]: Nice

File: HoloStorageAccessor/internal/holostorageaccessor/api_default_test.go [hunk 4]
```diff
@@ -0,0 +1,1470 @@
...
+			inMethod:   http.MethodGet,
+			wantStatus: 404,
+			wantErr: Error{
+				ErrorCode:    "404",
+				ErrorMessage: `id 'a3000' cannot be found`,
+			},
+		},
+	}
+
+	// Start a local HTTP server and Router
```
Comment (nielsboecker) [inline HoloStorageAccessor/internal/holostorageaccessor/api_default_test.go]: I don't understand why you do this eight times?

**Review by nielsboecker — APPROVED** [review]
Looks great and the test coverage now really provides some serious confidence.

The only thing I'm not sure about is that `api_default_test.go` is really long. Maybe it could be broken apart by main routes, e.g. `api_default_holograms_test.go`. The mock FHIR server could be its own file and for each test, only the relevant routes need to be mocked with return values.

We could then also consider to put all of that stuff in a `test` folder, even if it makes imports a bit harder and breaks the golang tradition of keeping it flat? To make it more concise you could even think about extracting the expected JSONs into actual `json` files.

But having said that I'm not sure if it is a good idea as I don't know the language, and I don't think you should put much more time into it necessarily.

File: HoloStorageAccessor/internal/holostorageaccessor/api_default_test.go (hunk 1 above)
Comment (boonwj) [reply HoloStorageAccessor/internal/holostorageaccessor/api_default_test.go]: shifted it to a separate file `mock_server_test.go`

File: HoloStorageAccessor/internal/holostorageaccessor/api_default_test.go (hunk 2 above)
Comment (boonwj) [reply HoloStorageAccessor/internal/holostorageaccessor/api_default_test.go]: absolutely

File: HoloStorageAccessor/internal/holostorageaccessor/api_default_test.go (hunk 4 above)
Comment (boonwj) [reply HoloStorageAccessor/internal/holostorageaccessor/api_default_test.go]: Changed it such that `TestMain()` function does an initial setup before running all tests.

**Comment by boonwj** [comment]
Addressed most of the concerns except for shifting the code out of the flat structure and moving the test input to jsons.

 Also found out that I have some bloated code that was removed.

**Review by nielsboecker — APPROVED** [review]
Double-approve

**Pull request merged**